internal/storage/bucket_handle_test.go

```diff
@@ -185,3 +186,29 @@ func (t *BucketHandleTest) TestStatObjectMethodWithMissingObject() {
 
 	AssertTrue(errors.As(error, &notfound))
 }
+
+func (t *BucketHandleTest) TestCopyObjectMethodWithValidObject() {
```

Comment:
is it possible to add tests for generation and metageneration validations?

Reply:
Added for generation. metageneration is not a part of fakeobject so it is not validating.